src/datumaro/components/environment.py

```diff
@@ -320,12 +232,16 @@ def detect_dataset(
         ignore_dirs = {"__MSOSX", "__MACOSX"}
         all_matched_formats: Set[DetectedFormat] = set()
 
+        extensions = get_all_file_extensions(path, ignore_dirs)
+
+        importers = {
+            importer
+            for extension in extensions
+            for importer in self.importers.extension_groups.get(extension, [])
+        }
```

Comment:
I am sorry for the confusion. If I correctly understood, all extensions are going to be investigated as a candidate for importers. But what we want do have is pre-filtration of candidate importers by feeding a specific extension to self.importers. For instance, we first detect extension within a dataset directory, and detect format later through 
```
extension = detect_extension(path)
importers = {
            importer
            for importer in self.importers.extension_groups.get(extension, [])
        }
 ```
 
 The main purpose of this PR is to improve the format detection performance.
 What do you think?

Reply:
Sorry, I misunderstood `get_all_file_extensions`. It looks working as expectation. Do you have any benchmark results between before PR and after PR? It must be great if we have :)

Reply:
Hi @itrushkin, could you provide any benchmark results when using `detect`?

Reply:
Excluding lazy plugin loading, calling `Environment.detect_dataset` 100 times yielded the following average times:
| `develop` | This PR |
| --- | --- |
| 0.035s | 0.009s |

**Note:** I excluded lazy plugin loading because importing `tensorflow` during some importer initialization takes about 5 seconds. I'm exploring ways to access importer file extensions without loading them entirely.